src/GraphQL/TypeExtensions.cs

```diff
@@ -129,20 +129,23 @@ public static Type GetGraphTypeFromType(this Type type, bool isNullable = false)
                 }
             }
 
-            var graphType = GraphTypeTypeRegistry.Get(type);
+            Type graphType;
 
             if (type.IsArray)
             {
-                var elementType = GetGraphTypeFromType(type.GetElementType(), isNullable);
-                var listType = typeof(ListGraphType<>);
-                graphType = listType.MakeGenericType(elementType);
+                var clrElementType = type.GetElementType();
+                var elementType = GetGraphTypeFromType(clrElementType, clrElementType.IsNullable()); // isNullable from elementType, not from parent array
+                graphType = typeof(ListGraphType<>).MakeGenericType(elementType);
             }
-
-            if (IsAnIEnumerable(type))
+            else if (IsAnIEnumerable(type))
+            {
+                var clrElementType = GetEnumerableElementType(type);
+                var elementType = GetGraphTypeFromType(clrElementType, clrElementType.IsNullable()); // isNullable from elementType, not from parent container
+                graphType = typeof(ListGraphType<>).MakeGenericType(elementType);
+            }
+            else
             {
-                var elementType = GetGraphTypeFromType(type.GenericTypeArguments.First(), isNullable);
```

Comment:
There was an error in the case of `IEnumerable` property (not `IEnumerable<T>`) - `GenericTypeArguments` returns empty array.